### DISCUSSION TITLE
Centralized users list for multi servers

### DISCUSSION THREAD

**Discussion by realbitt** [strongswan/strongswan#815] [General]
I have 3 servers , is there any way to centralized users list (one users list) instead of copy the same list for 3 servers.

**Comment by tobiasbrunner** [comment]
There are several ways to do that, e.g. using a central RADIUS server to authenticate client credentials via [_eap-radius_ plugin](https://wiki.strongswan.org/projects/strongswan/wiki/EapRadius), storing the credentials in a database via [_sql_ plugin](https://wiki.strongswan.org/projects/strongswan/wiki/SQL), using [VICI](https://wiki.strongswan.org/projects/strongswan/wiki/Vici) via TCP to load the credentials on all servers (could be risky to expose VICI via TCP, though), or use some file sharing or provisioning mechanism to distribute config files to all servers.

**Reply by realbitt** [reply]
database is good choice for me,  but when I connect all servers to same database I think all servers will load same configurations as well , while configurations other than users not the same for all servers.

**Reply by tobiasbrunner** [reply]
You don't have to store the config in the database. Just storing usernames (`identities` table) and passwords (`shared_secrets` table, associated via `shared_secret_identity` table) is possible, too.

**Reply by realbitt** [reply]
Thank you very mush, I will try it.

**Reply by realbitt** [reply]
how to enable --enable-sql for running instance, or I need to download source and configure it. ?

**Reply by tobiasbrunner** [reply]
Note that you also need either the _mysql_ or the _sqlite_ plugin. How to enable them depends on the distribution, there might be additional packages you have to install, or if that's not an option (e.g. Debian does not package these plugins), install from sources (see [this FAQ entry](https://wiki.strongswan.org/projects/strongswan/wiki/FAQ#Plugin-is-missing)).  If you install from sources, make sure to remove all binary packages before to avoid any conflicts.

**Comment by realbitt** [comment]
I canceled mysql database idea for users , I have used NPS(Radius in windows server ) for centralized users, it's very excellent and works very nice as exactly I want .

Thank you for guide me. @tobiasbrunner
